### PR TITLE
#98 Fix/route user to profile settings

### PR DESCRIPTION
**Pull request by deimonbody**

### CODE REVIEW

File: backend/src/services/user.service.ts [hunk 1]
```diff
@@ -209,16 +211,45 @@ export const registerCommonUsers = async (
 export const fetchUser = async (id: User['id']): Promise<UserWithRole> => {
   const userRepository = getCustomRepository(UserRepository);
   const roleRepository = getCustomRepository(UserRoleRepository);
+  const userQuizRepositiore = getCustomRepository(User_QuizCategoryRepository);
```
Comment (AndriyCh1) [inline backend/src/services/user.service.ts]: fix name of "userQuizRepositiore"

File: frontend/src/components/sidebar/sidebar.tsx [hunk 2]
```diff
@@ -8,13 +8,12 @@ import { ReactComponent as MenteeProfileIcon } from 'assets/img/icons/sidebar-ic
 import { MentorMenteeRoute } from 'common/enums/enums';
 import { Link } from 'components/common/common';
 import './styles.scss';
-import { useAppSelector } from 'hooks/store/store.hooks';
-
+import { useAppSelector } from 'hooks/hooks';
+import _ from 'lodash';
```
Comment (AndriyCh1) [inline frontend/src/components/sidebar/sidebar.tsx]: remove import if you don`t use it

**Review by AndriyCh1 — APPROVED** [review]

File: frontend/src/components/opportunities-page/opportunitiesPage.tsx [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import OpportunityPageItem from './opportunitiesPageItem';
```
Comment (nikita-remeslov) [inline frontend/src/components/opportunities-page/opportunitiesPage.tsx]: fix file name

File: frontend/src/components/profile-settings/steps/second-step.tsx [hunk 4]
```diff
@@ -1,31 +1,34 @@
...
     <div className="stepper__form">
       <WorkQuiz />
-      <StepControl
-        isValid={true}
-        isDisablePrevious={isDisablePrevious}
-        onPrevious={onPrevious}
-        onNext={onNext}
-        onSubmit={onSubmit}
-      />
+      {isCompleteTest ? (
```
Comment (nikita-remeslov) [inline frontend/src/components/profile-settings/steps/second-step.tsx:21]: Do we need 2 ternary operators?

File: frontend/src/components/profile/profile.tsx [hunk 5]
```diff
@@ -1,14 +1,29 @@
...
 import ProfileMain from './profile-main/profile-main';
 import './styles.scss';
-
 const ProfileInfo: React.FC = () => {
   const dispatch = useAppDispatch();
   const { user, isLoading } = useAppSelector((state) => state.profile);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (user) {
```
Comment (nikita-remeslov) [inline frontend/src/components/profile/profile.tsx]: try to simplify this logic. Very hard to understand what is going on here. You use `if` then `ternary operator` then `ternary operator`

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts @deimonbody

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**